lib/src/wso2/wso2-api-subscription/wso2-api-subscription.ts

```diff
@@ -0,0 +1,122 @@
+import { Construct } from 'constructs';
+import { CustomResource, RemovalPolicy } from 'aws-cdk-lib/core';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import z from 'zod';
+
+import { addLambdaAndProviderForWso2Operations } from '../utils-cdk';
+
+import { Wso2ApiSubscriptionProps } from './types';
+
+/**
+ * WSO2 API CDK construct for creating a WSO2 subscription from one application to an API
+ * This construct is related to one "physical" subscription in WSO2.
+ *
+ * The internal implementation tries to protect itself from various scenarios where larger or more complex
+ * WSO2 clusters might lead to out-of-order or delays in operations that happen asynchronously after the API
+ * accepts the requests, so for every mutation, there is a check to verify sanity.
+ *
+ * @example
+ *
+ * const wso2Api = new Wso2Api(...);
+ * const wso2Application = new Wso2Application(...);
+ *
+ * const wso2ApiSubscription = new Wso2ApiSubscription(this, 'Wso2ApiSubscription', {
+ *  apiId: wso2Api.wso2ApiId,
+ *  applicationId: wso2Application.wso2ApplicationId,
+ * });
+ *
+ * ---
+ *
+ * const wso2ApiSubscription = new Wso2ApiSubscription(this, 'Wso2ApiSubscription', {
+ *  apiSearchParameters: {
+ *    name: 'MyApi',
+ *    version: 'v1',
+ *    context: 'my-api',
+ *  },
+ *  applicationSearchParameters: {
+ *    applicationName: 'MyApplication',
+ *  },
+ * });
+ */
+export class Wso2ApiSubscription extends Construct {
```

Comment:
I was wondering about this taxonomy..

At first I thought about `api-subscription` to be very explicit, but internally in wso2 it is only called `subscription`.
In your draft #60 you also called it `subscription`.

Shall we change the taxonomy to be `subscription`?